edb/server/tenant.py

```diff
@@ -445,19 +453,27 @@ def create_task(
         if self._accept_new_tasks and self._task_group is not None:
             current_tenant.set(self.get_instance_name())
             if interruptable:
-                rv = self.__loop.create_task(coro)
+                rv = self.__loop.create_task(coro, name=name)
             else:
-                rv = self._task_group.create_task(coro)
+                rv = self._task_group.create_task(coro, name=name)
 
             # Keep a strong reference of the created Task
-            self._tasks.add(rv)
-            rv.add_done_callback(self._tasks.discard)
+            if name is not None:
+                self._named_tasks[name] = rv
```

Comment:
If we're sure about the uniqueness of task names, I'd suggest:

```suggestion
                assert name not in self._named_tasks
                self._named_tasks[name] = rv
```

to expose potential surprises in tests.